providers/implementations/macs/blake2_mac_impl.c

```diff
@@ -206,33 +200,38 @@ static int blake2_mac_set_ctx_params(void *vmacctx, const OSSL_PARAM params[])
         BLAKE2_PARAM_SET_DIGEST_LENGTH(&macctx->params, (uint8_t)size);
     }
 
-    if ((p = OSSL_PARAM_locate_const(params, OSSL_MAC_PARAM_KEY)) != NULL
-            && !blake2_setkey(macctx, p->data, p->data_size))
-        return 0;
+    if (p.key != NULL)
+        if (p.key->data_type != OSSL_PARAM_OCTET_STRING
```

Comment:
Why are these OCTET_STRING type checks needed?

Reply:
The parameter is an octet string.  If the existing code is followed through, there is no check for this.  I think it should have and we do type check params just about everywhere else.